### PR TITLE
Paper: Voice Capture, Structured Extraction, and Multi-Source Routing: Building and Evaluating a Field-Inspection Assistant in Python

### PR DESCRIPTION
**Pull request by CarolynOlsen**
Draft submission for 2026. Thank you for your time and consideration!

### CODE REVIEW

File: papers/carolyn_olsen/myst.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+version: 1
+extends: ../papers.yml
+project:
```
[the remark this answers is not in the document]
Comment (CarolynOlsen) [reply papers/carolyn_olsen/myst.yml:3]: I'm wondering if this is a false positive in the copilot auto PR review comments? I don't see any leading spaces here, unless I'm looking in the wrong place.

File: papers/carolyn_olsen/mybib.bib [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+	eprint       = {2407.11005},
+	archiveprefix = {arXiv}
+}
+@misc{fastapi,
+	title        = {{FastAPI}},
+	author       = {Ram{\'i}rez, Sebasti{\'a}n},
+	year         = 2018,
+	note         = {Accessed 2026},
+	howpublished = {\url{https://github.com/fastapi/fastapi}}
+}
```
[the remark this answers is not in the document]
Comment (CarolynOlsen) [reply papers/carolyn_olsen/mybib.bib:167]: Resolved in https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/98b6637c54c8018514b1221cff79e93b397875d2

File: papers/carolyn_olsen/main.md [hunk 3]
```diff
@@ -0,0 +1,593 @@
+---
+title: 'Voice Capture, Structured Extraction, and Multi-Source Routing: Building and Evaluating a Field-Inspection Assistant in Python'
+short_title: Building and Evaluating a Field-Inspection Assistant
+abstract: |
```
[the remark this answers is not in the document]
Comment (CarolynOlsen) [reply papers/carolyn_olsen/main.md:4]: Specified mybib.bib in https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/98b6637c54c8018514b1221cff79e93b397875d2

**Comment by CarolynOlsen** [comment]
Hello, @scipy-conference/2026-proceedings! I just wanted to make sure I didn't miss any steps for submission, since I haven't seen a reviewer assigned? Thank you in advance!

File: papers/carolyn_olsen/main.md [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+
+## The HiveGuide system
+
+HiveGuide has three components (@fig:arch): voice **capture** with structured
+extraction, **ingestion** of an authoritative corpus, and a multi-source
+**assistant**. The backend is FastAPI [@fastapi] on PostgreSQL [@postgresql] with
+the pgvector extension [@pgvector]; orchestration uses LangChain [@langchain]; the
+client is a React Native (iOS) app; and all model calls are routed through
+OpenRouter: GPT-4o for intent classification, `gpt-oss-120b` for generation, and
+`text-embedding-3-large` for vectors. These three lanes are the units a forker
```
Comment (ileri-oluwa-kiiye) [inline papers/carolyn_olsen/main.md:130]: The forkability argument is a strong part of the contribution. Consider adding a small table here distinguishing domain-agnostic infrastructure (capture/ingestion/assistant lanes, validation harness, routing interfaces) from beekeeping-specific components (inspection schema, corpus, routing examples). This would make the reuse story clearer for readers evaluating what they can directly adapt.

File: papers/carolyn_olsen/main.md [hunk 5]
```diff
@@ -0,0 +1,534 @@
...
+  - 0.388
+  - 0.583
+```
+
+**Classification accuracy tracks retrieval.** The supervised classifier reached
+97.8% accuracy and the best Hit@5 (19.5%), matching always-both while avoiding
+unnecessary retrieval. The embedding router achieved 87.0% with no model training
+(only labeled example queries), making it attractive when training data is scarce.
+Heuristics captured surprising signal (73.1%) from keyword patterns alone.
+
```
Comment (ileri-oluwa-kiiye) [inline papers/carolyn_olsen/main.md:468]: The relationship between routing accuracy and Hit@5 could use additional clarification here. The supervised classifier’s high routing accuracy does not translate into dramatically higher Hit@5 because retrieval quality remains a bottleneck after the correct source is selected. A short distinction between “selecting the correct source” and “retrieving the correct chunk from that source” would help readers interpret these metrics.

File: papers/carolyn_olsen/main.md [hunk 6]
```diff
@@ -0,0 +1,534 @@
...
+its annotated ground-truth chunks appears among the top-k retrieved chunks; retrieval
+ranks the corpus by cosine similarity and returns the top ten. Queries without
+chunk-level ground truth (most `personal_only` queries) are excluded from the hit-rate
+denominator. Response quality uses two **RAGAS** metrics (context relevance and
+response groundedness, 0-1) judged reference-free by Claude Haiku 4.5 at temperature
+0; context relevance is computed only for queries that retrieved document context.
+End-to-end **latency** is measured with `perf_counter` and includes retries, and
+**cost** combines per-model token usage at list prices for GPT-4o, `gpt-oss-120b`, and
+the Claude Haiku 4.5 judge.
+
```
Comment (ileri-oluwa-kiiye) [inline papers/carolyn_olsen/main.md:341]: The use of Claude Haiku 4.5 as an LLM judge is reasonable, but a brief justification would strengthen this section. For example, mention whether this choice was motivated by cost, availability, or prior evidence of agreement between LLM-based evaluation and human judgments. This is particularly important because groundedness scores drive one of the more surprising findings in the results.

File: papers/carolyn_olsen/main.md [hunk 7]
```diff
@@ -0,0 +1,534 @@
...
+        category_scores[category] = np.mean(top_k)
+    return max(category_scores, key=category_scores.get)
+```
+
+For parity, the remaining strategies are configured as follows. The **supervised**
+classifier trains a LightGBM model (`num_leaves=31`, `learning_rate=0.05`, 100
+boosting rounds) on TF-IDF features (unigrams and bigrams, up to 5000) of 499 labeled
+queries with a stratified 75/25 split. The **embedding** router indexes 20 example
+queries per category (60 vectors) and uses `k=3` neighbors, tuned on a held-out
+split. The **LLM** classifier uses GPT-4o with few-shot prompting; label
```
Comment (ileri-oluwa-kiiye) [inline papers/carolyn_olsen/main.md:311]: For reproducibility, could you specify the held-out split used for tuning (e.g., number of samples and how it was separated from the evaluation benchmark)?

**Review by ileri-oluwa-kiiye — APPROVED** [review]
## Overall Assessment

This is a strong, well-scoped submission. HiveGuide combines a genuine open-source Python contribution with an empirical evaluation of seven source-routing strategies on a 501-query benchmark. The paper is particularly strong in being transparent about limitations, including low absolute Hit@5 scores, agent failures, and single-pass evaluation constraints.

I recommend acceptance with minor revisions.

## Fit for SciPy

Clear yes. The contribution is concretely Python-based (FastAPI, pgvector, LangChain, Pydantic, LightGBM, scikit-learn, NumPy), reusable through its pluggable architecture, and framed as a domain-agnostic template rather than only a beekeeping application.

## Reproducibility

Strong. The three-command reproduction path, frozen results, fixed seed, checkpointing, and public repository provide a solid foundation for reproducibility.

## Writing Quality

The paper has a clear motivation, explicit contributions, minimal marketing language, and claims that are generally well-supported by experiments. The requested inline comments are primarily to improve clarity and reproducibility for readers.

**Comment by CarolynOlsen** [comment]
@ileri-oluwa-kiiye Thank you _so much_ for your comments and suggestions! I have made edits to address your feedback and will respond to your individual comments with the relevant commits.

File: papers/carolyn_olsen/main.md (hunk 7 above)
Comment (CarolynOlsen) [reply papers/carolyn_olsen/main.md]: This is now addressed here: https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/4bcb815068a784f73c91b888d3b76f04ebc84868

If the clarification isn't clear enough, please let me know.

File: papers/carolyn_olsen/main.md (hunk 6 above)
Comment (CarolynOlsen) [reply papers/carolyn_olsen/main.md:380]: Added explanation in https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/40e326169bbc9e11d9de09576dfab37799b4c727

File: papers/carolyn_olsen/main.md (hunk 5 above)
Comment (CarolynOlsen) [reply papers/carolyn_olsen/main.md:512]: Clarified in https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/6412ec095609fdcd709268b0e874f69379da86e6

File: papers/carolyn_olsen/main.md (hunk 4 above)
Comment (CarolynOlsen) [reply papers/carolyn_olsen/main.md:132]: Great idea -- I've added some text and a table here: https://github.com/scipy-conference/scipy_proceedings/pull/1222/commits/b95807d7a6a3208adbc966871a86834b30ab442b
